### PR TITLE
linode-api: init at 4.1.1b2

### PR DESCRIPTION
**Pull request by GlennS**
Added Linode's official Python library for their v4 API.

Note that this API is still in beta and subject to changes.

###### Motivation for this change

This should assist with adding Linode support to Nixops (see:
https://github.com/NixOS/nixops/issues/198).

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by GlennS** [comment]
Not sure what to do about the testing here, since this is my first pull request to Nixpkgs.

Can anyone advise on what else I need to do here?

**Comment by joachifm** [comment]
Please use sandboxing for testing, if you can.

File: pkgs/development/python-modules/linode-api/default.nix [hunk 1]
```diff
@@ -15,11 +15,16 @@ buildPythonPackage rec {
...
-  buildInputs = [];
   propagatedBuildInputs = [ requests ]
                              ++ stdenv.lib.optionals (!isPy3k) [ future ]
                              ++ stdenv.lib.optionals (pythonOlder "3.4") [ enum34 ];
 
+  postPatch = (stdenv.lib.optionalString (isPy3k) ''
+    sed -i -e '/"future",/d' setup.py
+  '') + (stdenv.lib.optionalString (!pythonOlder "3.4") ''
+    sed -i -e '/"enum34",/d' setup.py
+  '');
```
Comment (Mic92) [inline pkgs/development/python-modules/linode-api/default.nix]: The tests for python 3 still fail, while they are not executed for python2 (try building `pythonPackages3.linode-api`).

**Comment by Mic92** [comment]
Please disable check out the reason for failing tests: 
https://gist.github.com/Mic92/ebca4d7c604dce042e56cf6e02fab766

If it turns out that tests cannot be run inside a nix build sandbox due network restriction, you can disable them using `doCheck = false;` with a comment, why we cannot run the tests.

**Comment by GlennS** [comment]
I'm a little unclear about what's going on with the tests here. It doesn't look like the library actually has any tests at this point, so what was it that was failing?

**Comment by rvl** [comment]
Looks good - it appears to me that the module has no tests.

The failures from `unittest` seem to be from where it tries to import each module looking for tests.

I was trying it and:
1. It can import the module OK when I try with python 2.7, 3.4, 3.6.
2. I couldn't work out how to get an OAuth token for my Linode account. The option doesn't appear in my Linode settings ... would you have a link to some instructions?

Cheers

**Comment by GlennS** [comment]
The LinodeClient class doesn't actually need an OAuth token directly. You can give it a 'Personal Access Token' instead, and it will figure things out from there.

To get this you need to be on the profile section of Linode's new manager https://cloud.linode.com/profile/integrations/tokens

(You can see that there is also a tab for adding applications, which is more appropriate for someone who's using the library to make a web application.)

**Comment by rvl** [comment]
OK thanks, I just tested and it works for me.

**Comment by GlennS** [comment]
Is there anything more I need to do to get this moved along?

**Comment by rvl** [comment]
We just need someone with commit access to merge this PR.

It looks good to me.

**Review by LnL7 — APPROVED** [review]

**Comment by GlennS** [comment]
Thanks very much everyone for your help.

**Pull request merged**